src/main/java/application/FormplayerAuthFilter.java

```diff
@@ -38,26 +55,66 @@ public void init(FilterConfig filterConfig) throws ServletException {
     @Override
     public void doFilter(ServletRequest req, ServletResponse res,
                          FilterChain chain) throws IOException, ServletException {
-        HttpServletRequest request = (HttpServletRequest) req;
+        FormplayerHttpRequest request = new FormplayerHttpRequest((HttpServletRequest) req);
         if (isAuthorizationRequired(request)) {
-            if (!authorizeRequest(request)) {
+            // These are order dependent
+            if (getSessionId(request) == null) {
+                setResponseUnauthorized((HttpServletResponse) res);
+                return;
+            }
+            setToken(request);
```

Comment:
i tried doing that but i need a full instantiated request class in order to do these functions (i need to be able to get the POST). we'd have to make another method that would be initialize request or something which felt a little weird, but i'm not opposed to it